fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java

```diff
@@ -1405,7 +1405,15 @@ protected Function getBuiltinFunction(
             throws AnalysisException {
         FunctionName fnName = new FunctionName(name);
         Function searchDesc = new Function(fnName, argTypes, Type.INVALID, false);
-        return Catalog.getCurrentCatalog().getFunction(searchDesc, mode);
+        Function f = Catalog.getCurrentCatalog().getFunction(searchDesc, mode);
+        if (f != null && fnName.getFunction().equalsIgnoreCase("rand")) {
+            if (this.children != null
```

Comment:
`this.children != null` is unnecessary

Reply:
Rand function has two implements. One is Rand(const int), the other is Rand() without parm**.  this.children != null** filters Rand() without param.

Reply:
children is init with `Lists.newArrayList();` in `TreeNode`, so it won't be null.

Reply:
You are right.